packages/gafl-webapp-service/src/pages/terms-and-conditions/__tests__/route.spec.js

```diff
@@ -63,5 +73,98 @@ describe('terms-and-conditions > route', () => {
       )
       expect(data.paymentRequired).toBe(paymentRequired)
     })
+
+    it('test licenceTypeDisplay is called with permission and mssgs', async () => {
+      const func = () => getData(generateMockRequest({ [LICENCE_SUMMARY.page]: true, [CONTACT_SUMMARY.page]: false }))
+      await expect(func).rejects.toThrowRedirectTo(CONTACT_SUMMARY.uri)
+    })
+
+    const troutAndCoarse2Rods = 'Trout and coarse, up to 2 rods'
+    const troutAndCoarse3Rods = 'Trout and coarse, up to 3 rods'
+    const salmonAndSeaTrout = 'Salmon and sea trout'
+
+    it.each([[troutAndCoarse2Rods], [troutAndCoarse3Rods], [salmonAndSeaTrout]])(
+      'licenceTypeDisplay is called with permission and catalog',
+      async type => {
+        const permission = getMockPermission(10, type)
+        await getData(
+          generateMockRequest(
+            { [LICENCE_SUMMARY.page]: true, [CONTACT_SUMMARY.page]: true },
+            {
+              permissions: [permission]
+            }
+          )
+        )
+
+        expect(licenceTypeDisplay).toHaveBeenCalledWith(permission, catalog)
+      }
+    )
+
+    it.each([
+      [troutAndCoarse2Rods, true],
+      [troutAndCoarse3Rods, false],
+      [salmonAndSeaTrout, false]
+    ])('returns whether to display Trout and coarse, up to 2 rods conditions', async (type, displayType) => {
+      const mockLicenceTypeReturn = licenceTypeDisplay.mockReturnValue(type)
+      const mockLicenceTypeReturn3Rods = licenceTypeDisplay.mockReturnValueOnce(troutAndCoarse3Rods)
```

Comment:
It took me a while to get my head around what you were doing here and I'm still not sure I fully understand. `licenceTypeDisplay.mockReturnValue(type)` returns a function object, but you wouldn't normally hold it. You're then using this function, passing it to the getMockPermission as the permission type. `type` should be a string, but by doing this you're making it a function. So I'm not sure how the test is passing.

I think you should pass `type` to the first two calls on `getMockPermission`, then `troutAndCoarse3Rods` to the last call. Same goes for the other tests where you're doing this.